server/graphql/v2/mutation/AccountMutations.ts

```diff
@@ -48,6 +54,47 @@ const accountMutations = {
       });
     },
   },
+  addTwoFactorAuthTokenToIndividual: {
+    type: new GraphQLNonNull(Individual),
+    description: 'Add 2FA to the Account if it does not have it',
+    args: {
+      account: {
+        type: new GraphQLNonNull(AccountReferenceInput),
+        description: 'Account that will have 2FA added to it',
+      },
+      token: {
+        type: new GraphQLNonNull(GraphQLString),
+        description: 'The generated secret to save to the Account',
+      },
+    },
+    async resolve(_, args, req): Promise<object> {
+      if (!req.remoteUser) {
+        throw new Unauthorized();
+      }
+
+      const account = await fetchAccountWithReference(args.account);
+
+      if (!req.remoteUser.isAdminOfCollective(account)) {
+        throw new Forbidden();
+      }
+
+      const user = await models.User.findOne({ where: { CollectiveId: account.id } });
+
+      if (!user) {
+        throw new NotFound('Account not found.');
+      }
+
+      if (user.twoFactorAuthToken !== null) {
+        throw new Unauthorized('This account already has 2FA enabled.');
+      }
+
+      const encryptedText = crypto[CIPHER].encrypt(args.token, SECRET_KEY).toString();
+
+      await user.update({ twoFactorAuthToken: encryptedText });
```

Comment:
Don't we need to validate the token in the backend before persisting it?
I don't have any expertise in this so I'm not completely sure, it is just a gut feeling.
If we do, perhaps we should temporarily store cache it in Redis and when it is validated we move to the DB.

Reply:
That's a good point, we should do some basic validations on it (length, format...etc). We may also want to make sure that it's unique.

Reply:
This is a good idea - I will add some validation!

Reply:
Have added length/format validation. Would be interested in discussing best ways to check uniqueness as a further improvement next PR.